Sources/Configuration/Sentry.xcconfig

```diff
@@ -61,3 +61,6 @@ MODULEMAP_FILE = $(SRCROOT)/Sources/Sentry/Sentry.modulemap
 // Although `YES` is the default for iOS targets, command-line output via `xcodebuild -showBuildSettings`
 // will not emit this value unless it is explicitly set.
 SUPPORTS_MACCATALYST = YES
+
+// settings for Specto C++ port
+CLANG_CXX_LANGUAGE_STANDARD = c++17
```

Comment:
`h`: I'm not an expert on setting up C++ with Xcode. I guess adding the wrong settings here could mess up the SDK easily. Are we 100% sure this is the right setting? @brustolin can you maybe double-check?

Reply:
I don't know much about C++ dialect versions. 
The previous one was GNU C++ 11. 
Project looks to be compiling fine, maybe someone from the native team could tells us if any GNU C++ extensions was used in the existing code.

Reply:
IIRC we don't use any C++17 specific features so we could roll this back if needed, but it would be good to use a more modern standard regardless if it doesn't break anything.

Reply:
The one feature we use that requires a minimum of C++14 is `std::make_unique` here: https://github.com/getsentry/sentry-cocoa/blob/armcknight/specto-cpp-port/Sources/Sentry/Profiling/ThreadHandle.mm#L33

Reply:
@bruno-garcia let me know what you want to do here, we need C++17 for some features being used in this code (like std::optional), as far as I can tell this has no adverse effects as long as the SDK builds & tests. The only problem would be if we're trying to build this on machines that don't support a C++17 toolchain, but I don't think that applies, both local dev machines and CI should be OK.

Reply:
We provide the SDK with sources too so users can compile with their app. We just need to make sure this isn't going to introduce a breaking change or if it does, we should properly document and bump a major.

For example, does it affect cocoapods? 
![image](https://user-images.githubusercontent.com/1633368/157973838-5cfdccce-fa35-4824-a631-71887a6ac6a9.png)
(from [Stackoverflow](https://stackoverflow.com/questions/51288230/how-do-i-enable-c17-in-xcode-for-mac-osx))

Reply:
I think C++ 17 is probably fine. In this case I would propose we go ahead and require it and if it turns out that this breaks someone we figure out how to resolve this. I expect this would only be an issue on macOS and not iOS.

My only worry is that someone has their own build systems in place to build this and now need to pass this as well. This has been an issue in the past with Carthage and I can see similar issues if someone builds with Bazel etc.

Reply:
@mitsuhiko thank you, makes sense

@bruno-garcia it seems like this should definitely be a major then, especially with the other change we discussed on `SentryTransport`

Reply:
We have some validations for Carthage in CI https://github.com/getsentry/sentry-cocoa/blob/07d46d84d4435be652e05fb451c4b47c886c62aa/.github/workflows/build.yml#L68-L96

So we would know if this PR breaks Carthage.